packages/accordions/src/elements/timeline/components/Item.tsx

```diff
@@ -0,0 +1,52 @@
+/**
+ * Copyright Zendesk, Inc.
+ *
+ * Use of this source code is governed under the Apache License, Version 2.0
+ * found at http://www.apache.org/licenses/LICENSE-2.0.
+ */
+
+import React, {
+  useMemo,
+  forwardRef,
+  Children,
+  ReactNode,
+  ReactElement,
+  HTMLAttributes
+} from 'react';
+import { Timeline } from '../Timeline';
+import { StyledTimelineItem } from '../../../styled';
+import { TimelineItemContext, useTimelineContext } from '../../../utils';
+
+export interface IItem extends HTMLAttributes<HTMLLIElement> {
+  /** Defines the icon rendered in place of the dot */
+  icon?: ReactNode;
+  /** Provides surface color for an SVG icon placed on a non-white background */
+  surfaceColor?: string;
+}
+
+export const Item = forwardRef<HTMLLIElement, IItem>(({ icon, surfaceColor, ...props }, ref) => {
+  const value = useMemo(() => ({ icon, surfaceColor }), [icon, surfaceColor]);
+  const { isAlternate } = useTimelineContext();
+  let hasOppositeContent = false;
+
+  Children.forEach(props.children, child => {
+    if (child) {
```

Comment:
This is probably a redundant, if the component has no children then the [forEach short circuits and returns null](https://github.com/facebook/react/blob/aecb3b6d114e8fafddf6982133737198e8ea7cb3/packages/react/src/ReactChildren.js#L229-L231) so the callback is never entered.

Reply:
This `if` check is intended to prevent runtime errors when users pass `undefined`, `false`, or `null` as children since there isn't a `type` property on these children.